### PR TITLE
Feature/integration with other screens

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
Combine TestresultAvialable Screen with request Consents and update

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-3952

## Screenshot
Simulator Screen Shot - iPhone 12 Pro - 2020-11-26 at 12 43 21
Simulator Screen Shot - iPhone 12 Pro - 2020-11-26 at 12 43 24
Simulator Screen Shot - iPhone 12 Pro - 2020-11-26 at 12 43 31
Simulator Screen Shot - iPhone 12 Pro - 2020-11-26 at 12 43 34
Simulator Screen Shot - iPhone 12 Pro - 2020-11-26 at 12 44 26
Simulator Screen Shot - iPhone 12 Pro - 2020-11-26 at 12 43 40

### CODE REVIEW

**Comment by marcussc** [comment]
Please add Label and Milestone :)

**Review by SwissMountainFrog — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/TestResultAvailable/TestResultAvailableViewModel.swift [hunk 1]
```diff
@@ -4,30 +4,50 @@
...
+	
 	let didTapPrimaryFooterButton: () -> Void
 	let presentDismissAlert: () -> Void
-
-	var dynamicTableViewModel: DynamicTableViewModel {
-		DynamicTableViewModel([
+	
+	@Published var dynamicTableViewModel: DynamicTableViewModel?
+	var currentState: String = AppStrings.ExposureSubmissionTestresultAvailable.consentNotGranted
+	var dynamicTableviewModelPublisher: Published<DynamicTableViewModel?>.Publisher { $dynamicTableViewModel }
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/TestResultAvailable/TestResultAvailableViewModel.swift:36]: You don't need to define this separate publisher, you can just use `$dynamicTableViewModel` as that is already a publisher for this published variable.

File: src/xcode/ENA/ENA/Source/Scenes/TestResultAvailable/TestResultAvailableViewModel.swift [hunk 2]
```diff
@@ -4,30 +4,50 @@
...
 	// MARK: - Internal
-
+	
 	let didTapPrimaryFooterButton: () -> Void
 	let presentDismissAlert: () -> Void
-
-	var dynamicTableViewModel: DynamicTableViewModel {
-		DynamicTableViewModel([
+	
+	@Published var dynamicTableViewModel: DynamicTableViewModel?
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/TestResultAvailable/TestResultAvailableViewModel.swift:34]: You could initialize this var with an empty DynamicTableViewModel to get rid of the optionality

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewModel.swift [hunk 3]
```diff
@@ -107,10 +95,23 @@ class ExposureSubmissionTestResultConsentViewModel {
...
+
 				)
 			)
 		}
 	}
-	
+
+	// MARK: - Private
+
+	private let consentSwitch = UISwitch()
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewModel.swift:106]: Sorry that I just notice it when you moved it. But a UISwitch in a ViewModel? O.o

Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewModel.swift:106]: Looks like we need an extra DynamicTableViewCell with separated view model here. Shall we create a new Task for it?

File: src/xcode/ENA/ENA/Source/Scenes/TestResultAvailable/TestResultAvailableViewController.swift [hunk 4]
```diff
@@ -58,7 +61,17 @@ final class TestResultAvailableViewController: DynamicTableViewController, ENANa
 	private func setupTableView() {
 		view.backgroundColor = .enaColor(for: .background)
 		tableView.separatorStyle = .none
-		dynamicTableViewModel = viewModel.dynamicTableViewModel
+	}
+
+	private func setupViewModel() {
+		viewModel.$dynamicTableViewModel.sink { [weak self] dynamicTableViewModel in
+			guard let self = self else {
+				Log.debug("Faile to get dynamic tableview model - return early")
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/TestResultAvailable/TestResultAvailableViewController.swift:69]: You could remove this guard and just use `self?.`, only if you want to 😄 But the log is outdated and should be removed 😁

**Review by nickguendling — APPROVED** [review]

**Pull request merged**